pkg/controller/ephemeraljob/ephemeraljob_controller.go

```diff
@@ -375,3 +388,20 @@ func (r *ReconcileEphemeralJob) updateJobStatus(job *appsv1alpha1.EphemeralJob)
 	klog.V(5).Infof("Updating job %s status %#v", job.Name, job.Status)
 	return r.Status().Update(context.TODO(), job)
 }
+
+func (r *ReconcileEphemeralJob) removeEphemeralContainers(job *appsv1alpha1.EphemeralJob) error {
+	targetPods, err := r.filterPods(job)
```

Comment:
`filterPods` will get all pods that **can add the ephemeral containers into it**. But you might have to get pods that **already added ephemeral containers by this job**.

Reply:
fixed